src/web/mjs/connectors/Cuutruyen.mjs

```diff
@@ -0,0 +1,132 @@
+        const req = new URL('/api/v1/mangas/' + mangaid, this.api);
+        const request = new Request(req, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return new Manga(this, mangaid, data.data.attributes.name.trim());
+    }
+    async _getMangas() {
+        const uri = new URL('/api/v2/mangas/recently_updated', this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pages = data._metadata.total_pages;
+        let mangaList = [];
+        for (let page = 1; page <= pages; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+    async _getMangasFromPage(page) {
+        const uri = new URL(
+            '/api/v2/mangas/recently_updated?page=' + page,
+            this.api
+        );
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.data.map((element) => {
+            return {
+                id: element.id,
+                title: element.name.trim(),
+            };
+        });
+    }
+    async _getChapters(manga) {
+        const uri = new URL('/api/v2/mangas/' + manga.id + '/chapters', this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.data.map((element) => {
+            return {
+                id: element.id,
+                title: 'Chapter ' + element.number,
+            };
+        });
+    }
+    async _getPages(chapter) {
+        const uri = new URL('/api/v2/chapters/' + chapter.id, this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pages = data.data.pages;
+        return pages.map((image) => {
+            const scrambleString = image.drm_data.replace(/\n/g, "\\n");
+            return this.createConnectorURI({
+                url: image.image_url,
+                scrambleString,
+            });
+        });
+    }
+    async _handleConnectorURI(payload) {
+        const request = new Request(new URL(payload.url));
+        const res = await fetch(request);
+        const blob = await res.blob();
+        const image = await createImageBitmap(blob);
+        const canvas = this._descramble(image, this._decodeScrambleString(payload.scrambleString));
+        const blobFinally = await this._canvasToBlob(canvas);
+        return this._blobToBuffer(blobFinally);
+    }
+    _descramble(image, scrambleArray) {
+        const imgWidth = image.width;
+        const imgHeight = image.height;
+        let canvas = document.createElement('canvas');
+        canvas.width = imgWidth;
+        canvas.height = imgHeight;
+        const context = canvas.getContext('2d');
+        const tranformationArray = scrambleArray;
```

Comment:
`  const tranformationArray = scrambleArray;`

Not sure that variable is useful, cant we directly use `scrambleArray` ?